File: mds-demo-assignment/DSCI_521_LAB_ORIENTATION_R/tests/q2.R
test = list(
  name = "q2",
  cases = list(
    ottr::TestCase$new(
      hidden = FALSE,
      name = NA,
      points = 1.0,
      code = {
        testthat::expect_true(is.function(block_size))
        testthat::expect_equal(block_size(c(1, 1, 2), 1), 2)
        testthat::expect_equal(block_size(c(1, 1, 2), 2), 1)
      }
    )
  )
)

File: demo-assignment/tests/q2.py
OK_FORMAT = True

test = {
    "name": "q2",
    "points": 1,
    "suites": [{
        "cases": [
            # The slope is negative: more staff, shorter waits. A student who has
            # swapped x and y gets +2.57 and fails here rather than in question 4.
            {"code": ">>> round(float(slope), 3) == -0.389\nTrue", "hidden": False},
            {"code": ">>> round(float(intercept), 3) == 5.854\nTrue", "hidden": False},
            {"code": ">>> round(float(mse), 4) == 0.018\nTrue", "hidden": False},
        ],
        "scored": True,
        "type": "doctest",
    }],
}
